### DISCUSSION TITLE
OpenAPI auto schemas using Form(...) and Body(...)

### DISCUSSION THREAD

**Discussion by zephir89** [fastapi/fastapi#8558] [Questions]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
class LoginForm(BaseModel):
    username: str
    password: str

    def __init__(self, username: str = Form(...), password: str = Form(...)):
        super().__init__(username, password)


@router.post("/auth/login", response_model=Tokens)
def login(
        form_data: LoginForm = Depends(),
        db: Session = Depends(deps.get_db)
):
    pass
```

### Description

Hi guys,
I have a question regarding the snippet in Example Code. If I add `= Depends()` to `form_data` parameter, an auto OpenAPI schema named `Body_login_auth_login_post` will be added to Schemas section of Swagger UI, instead, if I simply write `form_data: LoginForm` this does not happen. Why? And why an auto schema is created if I'm using my schema named `LoginForm`?

Thank you :)

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

3.9.5

**Comment by xgorn** [comment]
The Form() function will create a schema form data, and that schema you have is `Body_login_auth_login_post`

**Comment by YuriiMotov** [comment]
With `form_data: LoginForm` it will be JSON Body parameter. FastAPI will add the schema of `LoginForm` to `components` section and reference it in request body object.

With `form_data: LoginForm = Depends()` it's equal to `form_data: LoginForm = Depends(LoginForm.__init__)` that is actually `form_data: LoginForm = Depends(__init__(username: str = Form(...), password: str = Form(...))`.
There is no any `LoginForm` schema in this case and the schema of body request will be created (`Body_login_auth_login_post`) and referenced by request body object.
